storage/src/lib.rs

```diff
@@ -71,10 +71,13 @@ pub struct BlockHeaderWithHash {
 impl BlockHeaderWithHash {
     /// Create block header extensions from plain block header
     pub fn new(block_header: BlockHeader) -> Result<Self, MessageHashError> {
-        Ok(BlockHeaderWithHash {
-            hash: block_header.message_hash()?.try_into()?,
-            header: Arc::new(block_header),
-        })
+        let hash = if let Some(hash) = block_header.hash().as_ref() {
```

Comment:
@akoptelov 
what if we put this if to `BlockHeader` directly or thought some trait
```
impl BlockHeader {
 fn block_hash(&self) -> Result<BlockHash, ...> {...}
}
```

Reply:
@akoptelov 
and we would need this also for `pub struct Operation {`,

Reply:
Probably there is even better solution, like a separate node containing data and its hash, but that should be done after I change encoding stuff. If you don't mind, let's keep it as it for now.

Reply:
@bkontur The fix is far from perfect, and probably it is even better to postpone it after encoding is redesigned, to do that properly. And now we can just give this changes to @dkuehr so he can continue fuzzing.